### PR TITLE
xsimd: add version 11.1.0, update xtl, minor improvement

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **xsimd/***

- add version 11.1.0
- update xtl
- use `is_apple_os`
- use `add_subdirectory` in test_v1_package

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xsimd/all/conanfile.py [hunk 1]
```diff
@@ -25,20 +25,19 @@ class XsimdConan(ConanFile):
...
 
     def requirements(self):
         if self.options.xtl_complex:
-            self.requires("xtl/0.7.4")
+            self.requires("xtl/0.7.5")
 
     def package_id(self):
         self.info.clear()
 
     def validate(self):
```
Comment (RubenRBS) [inline recipes/xsimd/all/conanfile.py:37]: now that you're doing minor improvements, reading https://github.com/xtensor-stack/xsimd#dependencies shows that if using xtl, the min cppstd is 14. Should we add that fact in the validate?

**Review by RubenRBS — COMMENTED** [review]
Thanks!

File: recipes/xsimd/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/xsimd/all/conanfile.py:37]: I agree with you.
I'm goingo to try it.

Comment (toge) [reply recipes/xsimd/all/conanfile.py:37]: @RubenRBS 
I agree with you.
I am going to try it.
Thanks!

Comment (toge) [reply recipes/xsimd/all/conanfile.py:54]: fixed

**Review by RubenRBS — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**